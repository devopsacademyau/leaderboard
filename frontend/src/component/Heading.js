import React from "react"

const H1 = ({ children }) => (
  <h1 className="f2 fw6 pv1 pt4-ns pb3-ns">{children}</h1>
)
const Link = ({ url, children }) => (
  <a
    href={url}
    target="_blank"
    rel="noopener noreferrer"
    className="link ttc black-80 hover-blue"
  >
    {children}
  </a>
)

const Heading = ({ repository }) =>
  repository && (
    <H1>
      <Link url={repository.url}>{repository.name}</Link> Leaderboard
    </H1>
  )

export { H1, Heading, Link }
