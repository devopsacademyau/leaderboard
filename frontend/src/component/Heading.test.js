import React from "react"
import { Heading } from "./Heading"
import renderer from "react-test-renderer"

test("matches snapshot", () => {
  const repository = {
    name: "my-repo",
    owner: "my-owner",
    url: "https://github.com/my-owner/my-repo"
  }
  const tree = renderer.create(<Heading repository={repository} />).toJSON()
  expect(tree).toMatchSnapshot()
})
