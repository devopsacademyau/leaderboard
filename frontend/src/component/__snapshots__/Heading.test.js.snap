// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<h1
  className="f2 fw6 pv1 pt4-ns pb3-ns"
>
  <a
    className="link ttc black-80 hover-blue"
    href="https://github.com/my-owner/my-repo"
    rel="noopener noreferrer"
    target="_blank"
  >
    my-repo
  </a>
   Leaderboard
</h1>
`;
